logger.go

```diff
@@ -0,0 +1,51 @@
+package flagsmith
+
+import (
+	"log"
+	"os"
+)
+
+// Logger is the interface used for logging by flagsmith client. This interface defines the methods
+// that a logger implementation must implement. It is used to abstract logging and
+// enable clients to use any logger implementation they want.
+type Logger interface {
```

Comment:
Ah, interesting, we have to define this ourselves?

Reply:
Yep, we have to